src/lib/autoinstall/widgets/ask_dialog.rb

```diff
@@ -0,0 +1,434 @@
+        # @macro seeAbstractWidget
+        def init
+          if !@question.value.nil?
+            self.value = @question.value.to_s
+            return
+          end
+
+          if @question.default_value_script
+            default_from_script = run_script(@question.default_value_script)
+          end
+          self.value = default_from_script || @question.default.to_s
+        end
+
+        # Stores the widget's value in the question
+        #
+        # @macro seeAbstractWidget
+        def store
+          @question.value = value
+        end
+
+      private
+
+        def run_script(script)
+          script.create_script_file
+          script.execute
+        end
+      end
+
+      # CheckBox widget for a question
+      class CheckBox < CWM::CheckBox
+        include AskWidget
+
+        # @param question [Y2Autoinstall::Ask::Question] Question to represent
+        def initialize(question)
+          textdomain "autoinst"
+          @question = question
+        end
+
+        # @macro seeAbstractWidget
+        def opt
+          [:notify]
+        end
+
+        # @macro seeAbstractWidget
+        def init
+          if !@question.value.nil?
+            self.value = @question.value
+            return
+          end
+
+          if @question.default_value_script
+            default_from_script = run_script(@question.default_value_script)
+          end
+          self.value = default_from_script == "true" || @question.default == "true"
+        end
+      end
+
+      class InputField < CWM::InputField
+        include AskWidget
+
+        # @param question [Y2Autoinstall::Ask::Question] Question to represent
+        def initialize(question)
+          textdomain "autoinst"
+          @question = question
+        end
+
+        # @macro seeAbstractWidget
+        def opt
+          [:hstretch, :notify, :notifyContextMenu]
+        end
+      end
+
+      class PasswordField < CWM::CustomWidget
+        include AskWidget
+
+        # @param question [Y2Autoinstall::Ask::Question] Question to represent
+        def initialize(question)
+          textdomain "autoinst"
+          @question = question
+        end
+
+        # Sets the password value
+        #
+        # This method updates both password widgets
+        #
+        # @return [String]
+        def value=(val)
+          new_value = val.to_s
+          Yast::UI.ChangeWidget(Id(widget_id), :Value, new_value)
+          Yast::UI.ChangeWidget(Id(confirmation_widget_id), :Value, new_value)
+        end
+
+        # Returns the password value
+        #
+        # @return [String]
+        def value
+          Yast::UI.QueryWidget(Id(widget_id), :Value)
+        end
+
+        # @macro seeAbstractWidget
+        def opt
+          [:hstretch, :notify, :notifyContextMenu]
+        end
+
+        def contents
+          VBox(
+            Password(Id(widget_id), Opt(*opt), label),
+            Password(Id(confirmation_widget_id), Opt(*opt), "")
```

Comment:
this looks kind of strange to not have any label.

Reply:
Yes, I know, but I was unsure about which label to use. So I decided to keep the old behavior.